File: src/main/java/com/braggbnb101/service/PropertyImageService.java
package com.braggbnb101.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnb101.domain.PropertyImage;
import com.braggbnb101.dto.PropertyImageDTO;
import com.braggbnb101.dto.PropertyImageSearchDTO;
import com.braggbnb101.dto.PropertyImagePageDTO;
import com.braggbnb101.dto.PropertyImageConvertCriteriaDTO;
import com.braggbnb101.service.GenericService;
import com.braggbnb101.dto.common.RequestDTO;
import com.braggbnb101.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PropertyImageService extends GenericService<PropertyImage, Integer> {

	List<PropertyImage> findAll();

	ResultDTO addPropertyImage(PropertyImageDTO propertyImageDTO, RequestDTO requestDTO);

	ResultDTO updatePropertyImage(PropertyImageDTO propertyImageDTO, RequestDTO requestDTO);

    Page<PropertyImage> getAllPropertyImages(Pageable pageable);

    Page<PropertyImage> getAllPropertyImages(Specification<PropertyImage> spec, Pageable pageable);

	ResponseEntity<PropertyImagePageDTO> getPropertyImages(PropertyImageSearchDTO propertyImageSearchDTO);
	
	List<PropertyImageDTO> convertPropertyImagesToPropertyImageDTOs(List<PropertyImage> propertyImages, PropertyImageConvertCriteriaDTO convertCriteria);

	PropertyImageDTO getPropertyImageDTOById(Integer propertyImageId);







}

File: braggbnb101_ui/src/services/PropertyImageService.js
import http from "../http-common"; 

class PropertyImageService {
  getAllPropertyImages(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/propertyImage/propertyImages`, searchDTO);
  }

  get(propertyImageId) {
    return this.getRequest(`/propertyImage/${propertyImageId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/propertyImage?field=${matchData}`, null);
  }

  addPropertyImage(data) {
    return http.post("/propertyImage/addPropertyImage", data);
  }

  update(data) {
  	return http.post("/propertyImage/updatePropertyImage", data);
  }
  
  uploadImage(data,propertyImageId) {
  	return http.postForm("/propertyImage/uploadImage/"+propertyImageId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PropertyImageService();
